redfish/updateservice.go

```diff
@@ -58,13 +58,18 @@ func (updateService *UpdateService) UnmarshalJSON(b []byte) error {
 
 	// Extract the links to other entities for later
 	*updateService = UpdateService(t.temp)
-	updateService.FirmwareInventory = string(t.FirmwareInventory)
+	updateService.firmwareInventory = string(t.FirmwareInventory)
 	updateService.TransferProtocol = t.Actions.SimpleUpdate.AllowableValues
 	updateService.UpdateServiceTarget = t.Actions.SimpleUpdate.Target
 	updateService.rawData = b
 	return nil
 }
 
+// FirmwareInventory will get the firmware inventory from the service
+func (updateService *UpdateService) FirmwareInventory() (*SoftwareInventoryCollection, error) {
```

Comment:
This will make this call inconsistent with the pattern followed elsewhere in the library. Everywhere else where there is a collection of related objects, the library takes care of that internal collection mechanism and just returns the desired objects to the users.

An example of this pattern can be seen here:

https://github.com/stmcginnis/gofish/blob/df4eea3f71d6c35a6dc7094ea936e2fd4d4f2091/redfish/chassis.go#L409

The collections object is consistent across the various \*Collection objects, so we can treat them all the same. So this should be updated to return the set of FirmwareInventory objects. And firmwareinventory.go will just need to add the ListReferenced\* convention used with the other objects to extract each instance out of the collections, like what is done here:

https://github.com/stmcginnis/gofish/blob/df4eea3f71d6c35a6dc7094ea936e2fd4d4f2091/redfish/networkadapter.go#L258

Reply:
I agree with the changes requested by @stmcginnis to follow the library pattern.

Also, note that Redfish UpdateService "/redfish/v1/UpdateService/" has the links for "FirmwareInventory" and "SoftwareInventory":

```
{
    "@odata.context": "/redfish/v1/$metadata#UpdateService.UpdateService",
    "@odata.etag": "W/\"97FD6D18\"",
    "@odata.id": "/redfish/v1/UpdateService/",
    "@odata.type": "#UpdateService.v1_1_1.UpdateService",
    "Id": "UpdateService",

	...
	
	"FirmwareInventory": {
        "@odata.id": "/redfish/v1/UpdateService/FirmwareInventory/"
    }
	
	... 
	
	"SoftwareInventory": {
        "@odata.id": "/redfish/v1/UpdateService/SoftwareInventory/"
    }
}

```

Looking at Redfish schema, my understanding is that both links in UpdateService should return a list of SoftwareInventory.

Reply:
I have created a new PR to better explain the point here: https://github.com/stmcginnis/gofish/pull/123